core/src/main/java/io/apicurio/hub/api/codegen/pre/OpenApiTypeInliner.java

```diff
@@ -126,14 +126,15 @@ public void visitNotSchema(Oas30NotSchema node) {
 
     /**
      * Returns true if the given schema is a simple type (e.g. string, integer, etc).
+     * or is a primitive type that should be inlined
      * @param schemaDef
      */
     private boolean isSimpleType(OasSchema schemaDef) {
         if ("string".equals(schemaDef.type)) {
             return schemaDef.enum_ == null;
         } else {
             return "integer".equals(schemaDef.type) || "number".equals(schemaDef.type) ||
-                    "boolean".equals(schemaDef.type);
+                    "boolean".equals(schemaDef.type) || "array".equals(schemaDef.type);
```

Comment:
Just my 2 cents... I think it would be better to add another branch in `visitSchema` above rather than checking for an array here.

Reply:
Agreed.  I'll make the change.